src/win/thread.c

```diff
@@ -191,7 +191,13 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 
 
 uv_thread_t uv_thread_self(void) {
-  return (uv_thread_t) uv_key_get(&uv__current_thread_key);
+  void* self;
+  uv_once(&uv__current_thread_init_guard, uv__init_current_thread_key);
```

Comment:
Can we instead move this uv_once initialization to the uv_init() code path? That way we should be guarantee that the TLS slot is always allocated, and remove the uv_once checks everywhere else, which should have better performance.